src/middleware/jwtExpiredErrorHandler.js

```diff
@@ -0,0 +1,16 @@
+const { names: errNames, messages: errMessages } = require('../constants/errors')
+
+module.exports = (err, req, res, next) => {
+  if (err.name !== errNames.UNAUTHORISED_JWT) {
+    return next(err)
+  }
+
+  switch (err.message) {
+    case errMessages.EXPIRED_JWT:
+      res.clearCookie('token')
+      res.redirect(req.originalUrl)
```

Comment:
Not convinced. I think 403. Ie your middleware has explicitly aborted the request. Either the middleware should let them through, or 403 them. This feels hacky

Reply:
Yeah I agree, it was just we have issues with jwt-express.

We're not using it as an authenticater really, we're setting `credentialsRequired:false` so it should let them through whether they have a cookie or not, but should populate req.user if it can.

However, it throws an error if they have an invalid cookie, which seems annoying/gives two different results to 2 very similar circumstances:

- A non logged in user tries to authorize and app, gets redirected to login page
- A user with an expired cookie tries to authorize an app and they get an authorization error

I guess what we want it to do is if the cookie is invalid, it should ignore it and let them through without populating `req.user` (i.e. as though they didn't have a cookie at all) - they will then get redirected to login if it turns out they need to have `req.user` populated

Reply:
I guess for now we can just send a 403 error for now though, but this will be an error that an end user will see:

1. User logs in, gets a token.
2. Token expires
3. User goes to an app, app asks them to authorize on OTP, redirects them to `OTP/oauth/authorize` with correct query parameters.
4. This is a secure route, so will check the cookie
5. jwt-express will give the user back an error

Reply:
@des-des it is hacky but we can't change the jwt-express middleware package, and we def don't want to be rejecting requests with expired tokens, we just want to let them through without populating req.user